src/charm.py

```diff
@@ -155,11 +158,11 @@ def _on_start(self, event: ops.StartEvent) -> None:
 
     def _on_stop(self, event: ops.StopEvent) -> None:
         utils.stop_service()
-        self.unit.status = ops.ActiveStatus("Service stopped")
+        self.update_status()
```

Comment:
yeah sorry for being in this PR. I think it's less confusing if the update_status function takes care of the status updates. Let me know if you think otherwise.